### PR TITLE
[ Layer ] Multiple Input Dimensions @open sesame 10/28 14:06

### PR DESCRIPTION
**Pull request by jijoongmoon**
Current implementaion only take one input. In order to take multiple
input, input_dim / output_dim should be vector type.
This PR includes this fixes except about addition layer which requires
actual multiple input. This will be done consequtive PR.

**Self evaluation:**
1. Build test:	 [X]Passed [ ]Failed [ ]Skipped
2. Run test:	 [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: jijoong.moon <jijoong.moon@samsung.com>

### CODE REVIEW

File: nntrainer/include/layer_internal.h [hunk 1]
```diff
@@ -83,7 +83,10 @@ class Layer : public ml::train::Layer {
     trainable(trainable_),
     num_weights(0),
     num_inputs(1),
-    num_outputs(1) {}
+    num_outputs(1) {
```
Comment (zhoonit) [inline nntrainer/include/layer_internal.h:86]: I think we should get rid of those two, as vector has it's own size, this violates Single source of truth principle

Comment (zhoonit) [reply nntrainer/include/layer_internal.h:86]: I'll make an issue to deal separately

**Review by zhoonit — APPROVED** [review]
LGTM

**Review by kparichay — APPROVED** [review]
LGTM

**Pull request merged**